src/desktop/apps/artist/server.tsx

```diff
@@ -58,6 +59,8 @@ app.get(
         },
       })
 
+      cacheHtmlForPage({ status, key: req.url, html: layout })
```

Comment:
(Bear with me while I try to understand.) This seems to cache the response using different logic than the middleware's for _fetching_ from the cache. In particular, it looks like it might write to the cache even if the requesting user is logged in, which risks caching and returning authenticated or personalized data to unauthenticated or just wrong visitors. What am I missing?

Ideally the "middleware" would be able to completely wrap the response-generation, so it can apply its escape logic before and possibly serve from the cache, or write to the cache after the response is generated if not. Is this kind of thing not possible from express middleware?

Reply:
Yea, you're totally right, the part that 'sets' the cache should also go through those same checks. This is an oversight.

I also believe you can wrap the entire request stack in Express with the middleware, which would enable removing all of these individual calls and consolidating everything into the one middleware location. Very good call!

Reply:
Turns out there isn't a proper middleware-y way to fully wrap the request, in order accomplish a `before, ...request..., after` style chain, but I did refactor in this spirit which cleans it up.